src/formatDate.js

```diff
@@ -8,7 +8,57 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitElement = fromFormat[3];
+  const joinElement = toFormat[3];
+  const newDate = date.split(splitElement);
+
+  let formatAgan = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      formatAgan[0] = newDate[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      formatAgan[1] = newDate[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      formatAgan[2] = newDate[i];
+    }
+  }
+
+  let year = formatAgan[0];
+  let month = formatAgan[1];
+  let day = formatAgan[2];
+
+  if (toFormat.includes('YYYY') && year.length === 2 && year < 30) {
+    year = `${20}` + `${year}`;
+  }
+
+  if (toFormat.includes('YYYY') && year.length === 2 && year >= 30) {
+    year = `${19}` + `${year}`;
+  }
+
+  if (toFormat.includes('YY')) {
+    year = year.toString().slice(-2);
+  }
```

Comment:
The year conversion logic is scattered across multiple if blocks. It would be more efficient and readable to handle all year-related conversions in a single code block.